gslib/boto_translation.py

```diff
@@ -651,13 +652,15 @@ def _PerformSimpleDownload(self,
                              num_progress_callbacks=XML_PROGRESS_CALLBACKS,
                              headers=None,
                              hash_algs=None):
-    try:
+    argspec = inspect.getfullargspec(key.get_contents_to_file)
```

Comment:
Using `inspect` is generally an anti-pattern. Can you do a type check with `isinstance(key, SOME_TYPE)`?
If not, can you put a comment here explaining why `inspect` is necessary?

Reply:
Switched to isinstance...